src/main/java/io/cdap/delta/bigquery/StructuredRecordToJson.java

```diff
@@ -185,6 +207,31 @@ private static void writeSimpleTypes(JsonWriter writer, String name, boolean isA
     }
   }
 
+  private static String checkAndTrimToMaxSupportedPrecision(String strValue) {
+    Matcher matcher = LOGICAL_DATE_PATTERN.matcher(strValue);
+    if (matcher.matches()) {
+      if (matcher.groupCount() == LOGICAL_DATE_WITH_TIMEFRACTION_GROUP_COUNT) {
```

Comment:
nit: merge into single if ?

Reply:
I think we can just remove the groupCount check it seems to only depend on the pattern and we already have null check on L216